_transcripts/005-rust-1.38.0.md

```diff
@@ -0,0 +1,686 @@
+episode, it is also— this is one of the nice things about the Rust release
+cycle, is that you can have releases that are really exciting, and some that are
+smaller ones. But at least you're always making progress.
+
+__Ben__: I think in this one, the biggest, the most exciting feature for most
+folks is going to be, like, potential compilation time improvements, for
+projects with large crate graphs. They're calling this one "pipelined
+compilation," and it is a new feature. It is on by default, I believe. So if
+you're using the newest stable release, you should be using it already. Do you
+want to go into what this means, Jon, or what you know about this?
+
+__Jon__: Yeah, so this is pretty cool. I mean, this is a complaint that we hear
+a lot in the Rust world, of the compiler takes forever to compile my thing. And
+one way to at least help with this problem is to compile more things in
+parallel. Now previously, the Rust compiler would, or Cargo would compile each
+crate sort of separately. So if you had some crate C that depends on both A and
+B, it would fully compile A than fully compile— or fully compile A and B, and
+then it would fully compile C. But it's not— it turns out that's not really
+necessary. Instead, you can compile the metadata for A and B, and then you can
+compile C straightaway after, in parallel with compiling the sort of contents of
+A and B.
+
+__Ben__: But what is metadata? You can't just gloss over that.
+
+__Jon__: So metadata is stuff like figuring out the types, figuring out what the
+functions are without necessarily compiling the body of them, like, without
+running things like optimizations.
+
+__Ben__: So kind of like header files, almost.
+
+__Jon__: Almost sort of like header files, yeah. All you really need is, you
+need to know all of the external parts of the crate, because those are the
+things that the other crates are going to use. And then, you only really need
+the final optimized compiled artifacts at the end, when you need to link them
+together.
+
+__Ben__: Yeah, so previously, any Rust crate is its own compilation unit, and
+anything that depends on it has to wait for it to be done before it can start.
+But I mean, again, like, you don't need to know everything about a compiled
+crate to actually begin compiling. And so this, kind of, just lets us kind of
+shortcut the process.
+
+__Jon__: Yeah, you can think of this as, you don't need the assembly code for
+all of the crates you're depending on, in order to compile your crate. Because
+much of it will be function calls into some part of the code that is external to
+the current crate. And then this is where link time optimization comes in. Where
+at link time, you can then try to do optimization across crate boundaries.
+
+The other thing, that's a little neat, as we talked about back in 1.36, the "mem
+uninit" type landed. And this was to try to deal with some issues around
+incorrect use of `mem::uninitialized`.
```

Comment:
I'm not sure what this talk of "mem uninit" is.  Seems like this is talking about MaybeUninit?  If "mem uninit" is wrong, I'm not sure if it should be formatted as code.

Reply:
Yup, definitely intended to be `MaybeUninit` (which indeed landed in 1.36). Not sure whether the transcript should have the error or the correct version?

Reply:
I don't have any other ideas, unless we'd like to sort of break the 4th wall and insert:

_(editor's note: should be `MaybeUninit`)_

Your call.

Reply:
I like the idea of an editor's note.

Reply:
OK, done.